tools/fv_io.F90

```diff
@@ -709,22 +743,25 @@ subroutine  remap_restart(Atm)
 
        fname = 'INPUT/fv_core.res'//trim(stile_name)//'.nc'
        if (open_file(Fv_tile_restart_r, fname, "read", fv_domain, is_restart=.true.)) then
+          call fv_io_register_axis(Fv_tile_restart_r, numx=2, numy=2, xpos=xpos_2d, ypos=ypos_2d, numz=2, &
+               zsize=zsize_2d)
           if (Atm(1)%flagstruct%is_ideal_case) then
-             call read_data(Fv_tile_restart_r, 'u0', u0_r)
-             call read_data(Fv_tile_restart_r, 'v0', v0_r)
+             call register_restart_field (Fv_tile_restart_r, 'u0', u0_r, dim_names_4d_u, is_optional=.true.)
+             call register_restart_field (Fv_tile_restart_r, 'v0', v0_r, dim_names_4d_v, is_optional=.true.)
           endif
-          call read_data(Fv_tile_restart_r, 'u', u_r)
-          call read_data(Fv_tile_restart_r, 'v', v_r)
+          call register_restart_field (Fv_tile_restart_r, 'u', u_r, dim_names_4d_u, is_optional=.true.)
+          call register_restart_field (Fv_tile_restart_r, 'v', v_r, dim_names_4d_v, is_optional=.true.)
```

Comment:
These should not be optional existience variables.  Please look at the original mpp_io version to confirm.